### ISSUE TITLE
Return Mercury and Gemini to old Max R&D?

### ISSUE THREAD

**Issue by peyre** [raceintospace/raceintospace#216]
As you probably know, the CD version of BARIS changed the Max R&D for the Mercury and Gemini capsules.  The changes were:
Mercury 78 -> 82
Gemini 83 -> 86
However Vostok stayed at 76 and Voskhod at 80.

It always seemed to me that the change gave a sizeable edge to the American player.  Not only did it make your first flights noticeably safer, but it made Gemini a very good contender for the Moon landings.  Gemini now has a higher Max R&D than Soyuz!  BARIS was severely criticized when it came out for being too difficult, and I've always suspected that this change was made because most players chose the US, which seems a poor reason to advantage one side over the other.

So I've wondered if we should put Mercury and Gemini back to their floppy version levels.  Any thoughts?

**Comment by peyre** [comment]
One important reason I wanted to bring this up was that an 86% Max R&D makes Gemini almost unbeatable.  It's the best way (in terms of competitiveness) to win as the US, and when I play as the Soviets I see the computer taking Gemini to the Moon probably 2/3 of the time.  I think it's potentially upsetting to game balance.

**Comment by brian0H** [comment]
Agree adjustments are required to make the game/simulation expand. Some of the capabilities granted to Gemini and Voskhod in game are beyond what was possible. My comments are all based to a historical model game where not all things are equal or fair.

Mercury and Vostok seem to be given a fair approximation of there limitations.

Voskhod should be no more capable than Vostok except should allow for jamming in 3 people for that mission goal and load only 2 people to go for space walk goal. Docking Missions should be beyond its scope. All this should be done at a lower cost follow on program to Vostok.

Gemini is yes given to high a Max% R&D and should at best be allowed to do a Lunar Flyby. The competitor for Apollo was "Gemini II" program which should be open for pursuit to that is a follow on to Gemini that is less costly than Apollo but less capable.

Soyuz needs at least two options a 2 crew design, a 3 crew design and maybe a striped down 1 man design "Zond". Only within Soyuz group should the USSR program be able to pursue Docking missions and on to Lunar Missions. The "Zond" being allowed to Manned Lunar Pass mission.

The exact Max% R&D for each program should vary more as you suggest.

**Comment by rnyoakum** [comment]
While it may be more historically accurate, it's worth noting that a _lot_ of work will have to be done before we can consider adding any different capsule programs. That will require a complete overhaul of the game's internal data structures, and save games (Issue #21).

**Comment by peyre** [comment]
Ryan is correct; that would be a big deal to implement.  The spinoff game "Blast Off!" used a system where you would assemble crews when you assigned the mission, so you could (IIRC) fly just, say, two people on a Soyuz mission.  However, RIS uses fixed crews who stick together, even to the point where, if something happens to one member of a Primary crew, the whole crew is replaced with the Backup crew.

During development of BARIS, there were plans to have a Zond capsule.  If you had developed Voskhod and Soyuz, for 15MB you could convert the Voskhod to the Zond capsule, and it was a two-man vehicle.  Its Safety matched Soyuz's safety, from what I've heard.  There are still scattered references to it in the code, and we even have pictures for the Zond capsule (liftoff.abz.SVC7.png and rdfull.but.55.png).  If we did implement something like this, I would probably want to rename it to L1 for historical accuracy.

Mercury and Vostok have one significant inaccuracy: you shouldn't be able to do EVA with them.  However, removing that capability would make it so there's very little you can do with that capsule; if we removed it I would want to create another mission for it, maybe a Medical Space Tests mission or something like that, to give you a reason to fly more than two or three missions with the capsule.

Yes, Voskhod is much more capable than it was in real life.  As it is, the game has made it sort of a stand-in for Voskhod, L1, and to some extend two-man Soyuz missions.  I wouldn't necessarily be against restricting Voskhod more, so long as we dropped the price of purchase and research quite a bit.  I like that Gemini is an option for the Moon landing, since there were plans to use modified Gemini capsules for the Moon landing.

**Comment by brian0H** [comment]
I agree cost adjustments would need to be made for several programs and believe that several should have an interrelated cost. The addition of a new mission or 2 to offset reduction of Mercury and Vostok/Voskhod capabilities would be great.

examples: what i am setting up in a excel environment > hope to learn to do similar within this RIS setting and environment.
Vostok program cost 18, launch load 1000 > if Voskhod done program cost 3
Voskhod program cost 24, launch load 1200 > if Vostok done program cost 6
(^ these programs would have more restricted capabilities)
R7 (or A Series) program cost 20, launch lift 800 & raise unit cost to 5
Booster "V" Stage program cost 10, launch lift +200 & reduce unit cost to 2
(this sets up a two stage development to do manned launches)
Booster "S" Stage program cost 20, launch lift +800 & unit cost at 3 > if Booster "V" done program cost 10
Soyuz program cost 36, launch load 1500 
(^ still working out some of the what if programs and relationships at this point)

For the USA program will be adding in a early launch vehicle (Jupiter/Thor/Redstone) for early satellite race. Also upper booster stages Able/Agena, Centaur. I believe Titan cost and capabilities need to be rolled back some and the Saturn I should be set up to replace it roll.

For both USA and USSR would also like to open up the possibility to do Skylab/Salyut Stations.

**Comment by peyre** [comment]
The awkward thing about adding missions is that all missions are numbered, generally from simplest (orbital satellite) to most complex (lunar landings), and there are no breaks in the numbers where we could insert something in between.  That doesn't mean it couldn't be done (and I have thought of some other missions that would be neat additions, such as Joint Orbital, where two capsules fly close to each other as a prestige stunt, like Vostok 5/6), but it might be a lot of work for the programmers.

Ah, Skylab & Salyut.  Those are a little late in our period, but they would make a good replacement for the MOL, which really should be called something like Orbital Rendezvous since its flight profile is more like Soyuz 5/6 than the historical MOL proposal.

**Comment by peyre** [comment]
...Waait a minute - if the MOL's flight profile matches the Orbital Rendezvous mission of Soyuz 5/6, why not rename it to that?  In the future we might consider adding an MOL or Skylab for the Americans and a Salyut for the Soviets, but that might get pretty tricky to code on the back end.

Anyone think we should rename the MOL for now?  It doesn't bear any real resemblance to the historical MOL, and the Soviets never had such a proposal.

**Comment by brian0H** [comment]
That would be a wonderful repurposing and it would remove a military program from the NASA queue. Further the glossed over rebuild of Gemini by the US Airforce to the Gemini B would no longer be a factor. Getting the option somehow to build and pursue Skylab/Salyut would be a wonder.

**Comment by peyre** [comment]
If we were to do a space station option, it would be down the road, since it wouldn't be a trivial change.  But the way I envision it is, in MISC the US would get an MOL or Skylab vehicle, and the Soviets would get a Salyut one.  It would be cheap to prototype and research (since it's really only good for one mission).  The trouble I see is that if we did that, there would then be calls to add an option to fly missions to the space station, and that's a lot of additional changes, mostly new missions and all the logic behind them.

**Comment by brian0H** [comment]
The USSR R7 (or A Series) should have two upper stage booster options.
1) Vo Booster > required to get Vostok and the early Luna probes into orbit.
2) So Booster > required to launch Voskhod, Soyuz's 7K-0K / 7K-L1 and the mid Luna probes.
- for Soyuz 7K-L3 it looks as if it would require D1/UR500 Proton to launch as mass is some 50% greater than 7K-0K.
Keeping it simple is see two kickers that could be launched with the R7.
1) Blok-L > used for early Luna probes 1,2,3.
2) Blok-E > used for mid Luna and Venera/Mars probes.
Blok-D > would be a third kicker that would require at least a D1/UR500 Proton to launch.

For the USA it seems to me that Able, Agena, and Centaur kind of fall into both camps, used with some launch systems as a upper stage boosters for other launch systems as kickers for placement of equipment into higher orbits or for transit orbit to the moon or planets. Still trying to break down there numbers and keep it simple.

**Comment by peyre** [comment]
Whoa, I think we're getting into a bit of scope creep there.  I realize that a lot of bits like that could be added to RIS for historical accuracy.  BASPM adds a lot of that kind of thing for instance.  However, I think one of the great strengths of RIS is that it abstracts out a lot of those kinds of details.  Its "keep it simple" approach is part of what keeps me playing RIS and not BASPM.  Dealing with all the fiddly bits of rocket stages stands a good chance of sucking the fun out of the game for a lot of people, whereas a cruder dividing line of "rocket vs. rocket+booster" keeps it much more manageable to casual players.

I have many times thought "It's a shame I can't fly a stripped-down version of that rocket for less, since I'm not putting much on top of it this time", but if you start adding a lot of these kinds of things it could start to get out of hand.  I wonder, though, if we could model it somewhat.  Maybe if you're putting a very light payload on a rocket (say, a DM for a Joint DM test, or putting a Mercury on a Titan), the game could reduce the cost of the rocket to account for reduced fuel and stages used to lift the payload.  Like an Atlas lifting just a DM might cost 2MB instead of 3, and a Titan lifting just a Mercury might cost 10MB instead of 12.

**Comment by brian0H** [comment]
For myself RIS is already a more versatile than BASPM. I develop R7 then develop add the upper booster to be able to launch Soyuz. Where it goes off mark is that R7 can launch Vostok from the start where it should require a second stage to get to orbit but could do a sub-orbital without booster. In BASPM you repeatedly redevelop slight variations on the same system. 

This game play wise can allow Manned Orbital a couple years sooner that historical timeline and this often pushed Moon landing much closer. While on the unmanned side of the equation I do not seem to be able to do the early Impact / Flyby missions of late 50's as I need Titan/D-1 rocket to get to the moon.

USSR wise Kicker A would be Blok-L for Luna missions 1,2,3 and Sputnik probe on R7(A Series) with Kicker A should be able to do Impact / Flyby Moon mission > But I would suggest with a lower prestige award as quality of info gained was far less than what the Cosmos probe would provide.

USA wise starting with Atlas as first Rocket feels almost like a cheat when I play. If there where a 1 stage rocket (a sort of Redstone/Jupitor/Thor mash up) that is cheaper and easier (than Atlas) but can only do satellites for that early race. Then the Atlas maybe a bit more expensive to research then Manned missions would slip into 1960~61 instead of 1959~1960 time frames.

**Comment by peyre** [comment]
That brings to mind that I've never been very comfortable for the name "Kicker".  Is it historical?  I'm not sure about that.  Is it a way to use a generic name for all those upper stages so the game doesn't have to have a series of upper stages that each can only be used with one rocket?

Included in that, I wondered if the Kicker-C should be renamed to match the [historical proposal it's based on](http://www.astronautix.com/l/l3-1963.html).  Trouble with that of course is that if brought to production it almost certainly wouldn't have been called "L3-1963", and when I asked Erik Anderson about it, he recommending leaving it alone.

**Comment by rnyoakum** [comment]
The idea of adding a reduced-cost version of the rocket is currently problematic, since there is no way to distinguish between a base version and a low-cost version. The only way to currently model that would be through the VAB interface. Consider this case:

I. I have two missions - one a Saturn w/ (Mercury or Ranger), another a Saturn w/ Apollo & DM.
II. I autopurchase the cheaper rocket for Mission A, but exit without assigning hardware.
III. I go into the VAB for Mission B, and assign the Saturn rocket to that mission
IV. I go back to Mission A, repurchase the cheaper rocket for Mission A, and assign hardware.

I've bought two rockets for reduced price, but one should have been full price.

**Comment by peyre** [comment]
The VAB is what I was thinking of.  I don't know if it's worth doing though, since it wouldn't affect price all that much.  I was just throwing the idea out there.

**Comment by peyre** [comment]
Incidentally, I have an idea if we wanted to make the Soviet capsules more realistic.  If you have Vostok developed and have flown at least an Orbital with it, you could be given the option to upgrade it to Voskhod for 10 or 15MB.  This would convert the Vostok to a Voskhod capsule with two seats and a Duration D and EVA capability but no other abilities.  It would have to break all crews of course.

Instead of our current Voskhod we could have the L1 (or LOK) capsule, a two-seater with Soyuz's capabilities but maybe a max of Duration D (not sure about that).  We could make it either a separate, independent capsule program or we could tie one to the other.  That is, they would have identical prototype and research costs, and researching one would research the other so they'd both always have the same Safety.  Purchase price would be a little different.  If we didn't tie one to the other, there should be rather heavy tech transfer between them and maybe prototype cost for one would be cut in half if the other had been prototyped.  And Kicker-C would probably still work only for the full Soyuz.

I don't know what anyone might think of this idea.  I know it would be a big deal to implement so I'm not asking for it, just suggesting it as an idea that would work somewhat within the program's current constraints.

**Comment by brian0H** [comment]
The concept of upgrading sounds reasonable for Vostok to Voskhod and then using the current Voskhod program position for the Soyuz 7K-L1/Zond option would better represent the historical possibilities better.
Vostok to Voskhod upgrade at 10MB seems ok. Unit cost should maybe also go to 4 or 5 MB.
Soyuz 7K-L1 probably can keep Program cost at 24 MB, unit cost should go up to 7? MB and R&D to 3? MB.

Question > Mission numbering in the "space history" any possibility of having unmanned missions dropped from the sequence?

**Comment by peyre** [comment]
Do you mean the unmanned lunar pass and orbit?  Those are kind of useless missions, except they're the only way to dummy-test a kicker if you're using one.

I've often thought it would be good to do something to make those more useful: like maybe flying one of those could be a milestone for the lunar pass.

As for changing the numbered missions: it's super-easy to change their names, but changing the missions they represent is another thing altogether.  I don't know how they're put together, honestly.

**Comment by brian0H** [comment]
In reference to the numbering, if 3 unmanned Vostok missions are made and then go for a manned orbital well in space history it is Vostok #4. I was wondering if unmanned capsule missions could get labeled under a different name > say use Korabl unmanned missions so that the first manned Vostok mission will show up as #1.

**Comment by rnyoakum** [comment]
I'm issuing a warning before anyone starts trading out mission slots: replacing missions is a complicated issue that's not as simple as changing the missions file. For one, the game keeps track of the missions flown, and the expected mission steps would no longer match the history of those missions in save games. Effectively, using a save that recorded one of those missions would  analogous to using a corrupt save. So if adding new missions, I see three approaches:

1. If replacing existing missions, wait until version change and declare all previous saves "dubious."
2. Add the new missions onto the end of the missions file.
3. Convert the mission data system to a format that allows missions to be easily added.

Each of these presents their own challenges, which I'll discuss if anyone is interested.

**Comment by peyre** [comment]
You're thinking of effectively emulating the Soviet numbering system?  The thought has occurred, but as it is the game sort of emulates the US system.  It's simpler this way, and changing it would take a bit of programming.  Personally I'm happy with the way it is now, though I wouldn't say no if the programmers wanted to change it.  We would have to determine a whole set of alternative names though, and decide whether to have just the Soviets use an alternative set of names, or do it for both sides.

**Comment by peyre** [comment]
Thank you Ryan.  I knew that adding missions would be a difficult structural change on the back end, but I didn't know the details.  I've thought before about adding missions, but the ones I've thought to add are all early on, and adding missions would be best done at the end.  Plus, there's the complication about what we do with the last, hidden missions: the rescues.  I don't know if there's partial code still in the game for them that would have to be remmed out or something.

One thing I've thought about would be a medical/scientific milestone, to give you something else you can do with Mercury/Vostok in case we remove its EVA capability - and also something to give you a reason to fly at least one more Mercury/Vostok mission.  But maybe instead of a discrete mission, we could require at least two orbitals before you can move on to duration: some special system along the lines of the LM points system.  A possibility, but possibly more trouble than it would be worth.

**Comment by peyre** [comment]
I just had it reinforced to me how overpowered Gemini is.  I hadn't even flown Duration C (as the Soviets) when this happened:

![Gemini1967](https://user-images.githubusercontent.com/3118014/67164609-38004680-f331-11e9-9c1f-962a2c2b407a.png)

**Comment by peyre** [comment]
Getting back to the original topic, unless people think otherwise, I'd suggest it would be a good idea to put Mercury and Gemini back to their original (Floppy) values, 78 and 83% respectively.  Ryan, do you know what's involved in changing those kinds of values?

**Comment by rnyoakum** [comment]
Each piece of hardware has a MaxRD attribute associated with it, stored within the player data. For a saved game, this player data is preserved in the save file and loaded from the save file into memory. Consequently, existing saves will not be affected by a Max R&D change.

For new games, starting player data is set using the urast.dat file. It's a game data template - a byte array copied in order into the memory space of a player data variable (with endian correction). It's related to rast.dat, which is a compressed version of the same file and probably a predecessor (I'd have to check to see if they're still synced). The rast.dat version is no longer used, in deference to urast.dat. Being a byte array, there is no context embedded in the file as to what any of the data represents. That makes editing it tricky.

Short term, I can edit the values in urast.dat if you'd like. Longer term, the file should be converted to use a human-readable format (JSON is my plan).

**Comment by peyre** [comment]
Wow, thanks.  I don't want to distract you from any development work you might prefer to do, but it would be nice to change the values back (_unless_ others think they'd be better left alone).  Ideally I think it'd be nice to have an option in Advanced Preferences that lets you set the values back to the CD version values if you so desire, but it's not that important.  (Something like that might better wait for conversion of the files to JSON.)

**Comment by peyre** [comment]
Incidentally, I can't make heads or tails of the urast.dat file.  How would I change the R&D etc. values in it?  I don't see anything in there that corresponds to the stats for hardware, though I can see all the hardware names.

**Comment by rnyoakum** [comment]
The urast.dat file is an uncompressed version of a `struct Players`, initialized to the default values used at the start of the game. As you're aware, most of the game information is stored in the variable `Data` (declared in game_main.cpp). When urast.dat is read, the byte data from the file is read *directly into* the memory space reserved by `Data`. Therefore, the Max R&D value isn't recorded as a text entry, but as a single byte. So it's searching for a single byte in a file of bytes.

Currently, there is no easy way to modify the R&D values. Changing something in urast.dat entails looking at the struct definitions in data.h and calculating byte offsets, then modifying that byte with a hex editor. To modify the Gemini entry entails finding the Gemini equipment entry, which can be found by searching for "Gemini" in the file. The 'G' in Gemini is the first byte in the entry. For an equipment entry, the R&D value is the 49th byte. 

I'll take a look at coding a helper utility to convert urast.dat back and forth between a .dat file and a user-friendly JSON file. That's the sort of thing I probably should be doing next for RIS, it's just not going to be very fun...which is probably why I haven't done it already :(

**Comment by peyre** [comment]
Har, and probably why the original crew dropped the project back in '13.  Well I don't blame you for not looking forward to the task.  Frankly I'm less excited about improvements to the back end than I am in bug fixes and enhancements, though if you get motivated to do something on the back end I wouldn't try to discourage you.  Thanks for letting me know about the urast file; I'll sit tight on it for now then...or maybe, compare it to the old rast.dat file that the floppy version used.

**Comment by peyre** [comment]
FYI, I think the game is getting its data from somewhere else.  I found the bytes and changed them, then recompiled, but it doesn't change the values in the game.  I even tried changing "SATURN V" to just "SATURN" to see if it would take a name change, but that has no effect either.

And come to think of it, each component has different values for Max R&D and Max Safety: one for the US, one for the USSR, and one for either in the Basic Model.

**Comment by rnyoakum** [comment]
Ah, sorry. The game data **is** loaded from urast.dat, but that contains the basic model data. When the historical model is selected, the equipment entries are overwritten with data from hist.dat. That's the file you will have to modify. It's a much simpler file - it's still raw data, but only the equipment entries.

You don't have to worry about modifying multiple values; each component only has one Max R&D and one Max Safety value. Those values are initialized when a game begins, with the basic model values (US & USSR) found in urast.dat and the historical equipment (US & USSR) recorded in hist.dat. The game data records US & USSR equipment separately, so Gemini has its own data entry distinct from the Soviet equivalent.

For testing the modifications, any changes to the Max R&D / Max Safety will only apply to a new game - each save file stores the Max R&D / Max Safety of each component, and uses those values. Also, you won't need to recompile the game to test the changes. The files, and the values within, are read at runtime.

**Comment by peyre** [comment]
**Aha!**  And, as usual, it's something I really should have been able to work out myself.  Sure enough, that worked.

Incidentally, I've been toying with the idea of renaming the Saturn V to just the Saturn, to include the Saturn 1A and 1B.  After all, the Titan also represents a family of rockets; it isn't called "Titan III" or whatever.  Anyone have any thoughts on that?

**Comment by peyre** [comment]
Ok, I've played the US with these new values and Gemini is tougher, but still feasible.  I've gone ahead and committed the changes.  We can, of course, revisit this if people disagree with the changes.  In particular, I wonder if Apollo/Soyuz should get a higher Max R&D in the Basic version - 84% is lower than either Apollo or Soyuz in the Historical Model.

**Comment by rnyoakum** [comment]
We'll want to update the manual to reflect the new changes; I didn't see that in the commit.

**Comment by peyre** [comment]
Oh yes.  Also, as mentioned above, I'm thinking about restoring the Apollo/Soyuz Max R&D from the CD version.  But I'll have to make sure to update the manual before our next release.  How close to that do you think we're getting, btw?

**Comment by peyre** [comment]
I have gone ahead and restored the CD version value of the Basic Model Apollo/Soyuz Max R&D.  I really think it makes more sense than the floppy version value.

**Comment by peyre** [comment]
Something I've been thinking about since I made this change...there's one thing about it that's never sat well with me.  For most hardware, the Basic Model sets values in between the US and Soviet  The Max R&D for Apollo in between the two, or at the lower level.  So for instance Mercury/Vostok was set to 76% (the Soviet value), but Gemini/Voskhod was 81% (in between 80 & 83).  The Max R&D for Apollo, though was set to 84%, actually lower than either Apollo (89%) or Soyuz (85%)!  The CD version changed it to 87%, in between the two.  Maybe we should change Apollo/Soyuz back to the CD version.

**Comment by peyre** [comment]
Huh!  It looks like I did that already.  I must have (accidentally?) left Apollo alone in urast.dat.  Well . . . ok then!
